CMakeLists.txt

```diff
@@ -0,0 +1,397 @@
+cmake_minimum_required (VERSION 2.8)
+project (logcabin)
+set (LOGCABIN_VERSION_MAJOR 1)
+set (LOGCABIN_VERSION_MINOR 2)
+
+include_directories(BEFORE .)
+include_directories(BEFORE ./include)
+include_directories(BEFORE ./build)
+
+# c++ std should be defined in another way, but let make it work first
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-strict-overflow -fPIC")
+
+set(CMAKE_PROTO_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+set(CMAKE_GTEST_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g -DDEBUG")
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_R} -O2 -DNDEBUG")
+
+
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    message("using clang")
+    set (FLAG_LIST 
+        -Wno-c++98-compat-pedantic 
+        -Wno-covered-switch-default 
+        -Wno-deprecated 
+        -Wno-disabled-macro-expansion 
+        -Wno-documentation-unknown-command 
+        -Wno-exit-time-destructors 
+        -Wno-float-equal 
+        -Wno-global-constructors 
+        -Wno-gnu-zero-variadic-macro-arguments 
+        -Wno-missing-noreturn 
+        -Wno-missing-prototypes 
+        -Wno-missing-variable-declarations 
+        -Wno-packed 
+        -Wno-padded 
+        -Wno-reserved-id-macro 
+        -Wno-shadow 
+        -Wno-shift-sign-overflow 
+        -Wno-switch-enum 
+        -Wno-undef 
+        -Wno-unknown-warning-option 
+        -Wno-unused-macros 
+        -Wno-unused-member-function 
+        -Wno-unused-parameter 
+        -Wno-used-but-marked-unused 
+        -Wno-vla 
+        -Wno-vla-extension 
+        -Wno-weak-vtables
```

Comment:
looks like -Weverything got lost during the port. I don't recall why it's on a separate env.Prepend() call in Sconsfile.